### PR TITLE
[CodeQuality] Handle crash on yield from on OptionalParametersAfterRequiredRector

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://getrector.com/demo/a621ae26-28b5-4936-93e0-d3c13b3f9ad9

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by TomasVotruba** [comment]
Thank, looks good to me :+1:

File: src/Configuration/OnlyRuleResolver.php [hunk 1]
```diff
@@ -60,7 +62,7 @@ public function resolve(string $rule): string
             throw new RectorRuleNameAmbigiousException($message);
         }
 
-        if (strpos($rule, '\\') === false) {
+        if (in_array(str_contains($rule, '\\'), [0, false], true)) {
```
Comment (samsonasik) [inline src/Configuration/OnlyRuleResolver.php:65]: rectify seems should be only `str_contains($rule, '\\') === false`, as str_contains never return 0, only `bool`. I will check.

Comment (samsonasik) [reply src/Configuration/OnlyRuleResolver.php:65]: It seems due 2 rules, with union type, that buggy due to usage of:

```php
if ($exprType->isBoolean()->yes()) {
```

```
 0/1 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
PHPStan\Type\UnionType #22767
   types: array (2)
   |  0 => PHPStan\Type\IntegerRangeType #20331
   |  |  min: 0
   |  |  max: null
   |  1 => PHPStan\Type\Constant\ConstantBooleanType #22757
   |  |  value: false
   normalized: true
   sortedTypes: true
   cachedDescriptions: array (1)
   |  4 => 'int<0, max>|false'
```


```diff
➜  rector-src git:(main) ✗ bin/rector process src/Configuration/OnlyRuleResolver.php --dry-run
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
1 file with changes
===================

1) src/Configuration/OnlyRuleResolver.php:61

    ---------- begin diff ----------
@@ @@
             throw new RectorRuleNameAmbigiousException($message);
         }

-        if (strpos($rule, '\\') === false) {
+        if (in_array(str_contains($rule, '\\'), [0, false], true)) {
             $message = sprintf(
                 'Rule "%s" was not found.%sThe rule has no namespace. Make sure to escape the backslashes, and add quotes around the rule name: --only="My\\Rector\\Rule"',
                 $rule,
    ----------- end diff -----------

Applied rules:
 * StrContainsRector
 * BooleanInBooleanNotRuleFixerRector
```

Comment (samsonasik) [reply src/Configuration/OnlyRuleResolver.php:65]: That's seems issue in phpstan side for detecting it as union, I just directly patch it on the code for now:

- https://github.com/rectorphp/rector-src/pull/6546

**Pull request merged**